File: gdscript/src/main/java/gdscript/psi/GdFile.java
package gdscript.psi;

import com.intellij.extapi.psi.PsiFileBase;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.psi.FileViewProvider;
import gdscript.GdFileType;
import gdscript.GdLanguage;
import org.jetbrains.annotations.NotNull;

public class GdFile extends PsiFileBase {

    public GdFile(@NotNull FileViewProvider viewProvider) {
        super(viewProvider, GdLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public FileType getFileType() {
        return GdFileType.INSTANCE;
    }

    @Override
    public String toString() {
        return "GdScript File";
    }

}

File: gdscript/src/main/kotlin/gdscript/psi/GdFile.kt
package gdscript.psi

import com.intellij.extapi.psi.PsiFileBase
import com.intellij.openapi.fileTypes.FileType
import com.intellij.psi.FileViewProvider
import com.jetbrains.rider.godot.community.gdscript.GdFileType
import com.jetbrains.rider.godot.community.gdscript.GdLanguage

open class GdFile(viewProvider: FileViewProvider) : PsiFileBase(viewProvider, GdLanguage) {
    override fun getFileType(): FileType = GdFileType

    override fun toString(): String {
        return "GdScript File"
    }
}
